src/store/authorization/actions.ts

```diff
@@ -35,20 +39,46 @@ export const actions: ActionTree<AuthorizationStateInterface, RootStateInterface
         return payload.errorCallback(r);
       });
   },
-  [AuthActionEnum.GET_DATA]: ({ commit }, payload) => {
-    HTTP.get(`/users/${payload}`).then((r: AxiosResponse<UserInterface>) => {
+  [AuthActionEnum.GET_DATA]: async ({ commit }, payload) => {
+    await HTTP.get(`/users/${payload}`).then((r: AxiosResponse<UserInterface>) => {
       commit(AuthMutationEnum.SET_USER, r.data);
     });
   },
-  [AuthActionEnum.UPDATE_USER]: ({ commit }, payload) => {
-    HTTP.get(`/users/${payload}`).then((r: AxiosResponse<UserInterface>) => {
-      commit(AuthMutationEnum.SET_USER, r.data);
-    });
 
-    commit(AuthMutationEnum.SET_USER, null);
-  },
   [AuthActionEnum.SIGN_OUT]: ({ commit, dispatch }, payload) => {
     dispatch('localStorageStore/REMOVE', 'user', { root: true });
     commit(AuthMutationEnum.SET_USER, payload);
   },
+  /////////// MY ACTIONS
+  [AuthActionEnum.SET_FORM]: ({ commit }, payload) => {
+    commit(AuthMutationEnum.SET_FORM, payload);
+  },
+
+  [AuthActionEnum.UPDATE_USER]: async ({ state, commit }, payload: UpdateUserInterface) => {
+    const payloadData = new UpdateUserModel(payload);
+    const userId = state.user!.id;
+    await HTTP.put(`/users/${userId}`, payloadData).then((r: AxiosResponse<UserInterface>) => {
+      commit(AuthMutationEnum.SET_USER, r.data);
+    });
+  },
+
+  [AuthActionEnum.DELETE_CONTACT]: async ({ state, commit, dispatch }, payload) => {
+    const userId = state.user!.id;
+    await HTTP.delete(`/users/${userId}/contacts/${payload}`);
+    await HTTP.get(`/users/${userId}/contacts`).then((r: AxiosResponse<ContactInterface>) => {
```

Comment:
Why do you need two requests here?

Reply:
The second request needed to update the contact in store. Without this request, I could not update the state.user.contacts.

Reply:
You can update contact in store in mutation with response payload, that you will get on delete action from BE